source/winVersion.py

```diff
@@ -1,65 +1,186 @@
+		On Windows 10, pass in either '10' or the string representing a specific Windows 10 release.
+		For example, '20H2' will return Windows version 10.0.19042.
+		"""
+		if release == "7":
+			return WIN7 if servicePack == "" else WIN7_SP1
+		elif release == "8":
+			return WIN8
+		elif release == "8.1":
+			return WIN81
+		elif release in ("10", "1507"):
+			return WIN10
+		elif release in WIN10_RELEASE_NAME_TO_BUILDS:
+			return WinVersion(
+				major=10,
+				minor=0,
+				build=WIN10_RELEASE_NAME_TO_BUILDS[release]
+			)
+		else:
+			raise ValueError(f"Unknown Windows release: {release}")
+
+	@staticmethod
+	def fromVersionText(versionText: str):
+		"""Returns a Windows version information based on version string
+		of the form major.minor.build.
+		"""
+		major, minor, build = versionText.split(".")
+		# Specifically for Windows 7 service pack 1.
+		servicePack = "1" if versionText == "6.1.7601" else ""
+		return WinVersion(
+			major=int(major),
+			minor=int(minor),
+			build=int(build),
+			servicePack=servicePack
+		)
+
+
+# Windows releases to WinVersion instances for easing comparisons.
+WIN7 = WinVersion(major=6, minor=1, build=7600)
+WIN7_SP1 = WinVersion(major=6, minor=1, build=7601, servicePack="1")
+WIN8 = WinVersion(major=6, minor=2, build=9200)
+WIN81 = WinVersion(major=6, minor=3, build=9600)
+WIN10 = WIN10_1507 = WinVersion(major=10, minor=0, build=10240)
+WIN10_1511 = WinVersion(major=10, minor=0, build=10586)
+WIN10_1607 = WinVersion(major=10, minor=0, build=14393)
+WIN10_1703 = WinVersion(major=10, minor=0, build=15063)
+WIN10_1709 = WinVersion(major=10, minor=0, build=16299)
+WIN10_1803 = WinVersion(major=10, minor=0, build=17134)
+WIN10_1809 = WinVersion(major=10, minor=0, build=17763)
+WIN10_1903 = WinVersion(major=10, minor=0, build=18362)
+WIN10_1909 = WinVersion(major=10, minor=0, build=18363)
+WIN10_2004 = WinVersion(major=10, minor=0, build=19041)
+WIN10_20H2 = WinVersion(major=10, minor=0, build=19042)
+
+
+def getWinVer():
+	"""Returns a record of current Windows version NVDA is running on.
+	"""
+	winVer = sys.getwindowsversion()
+	return WinVersion(
+		major=winVer.major,
+		minor=winVer.minor,
+		build=winVer.build,
+		servicePack=winVer.service_pack,
+		productType=("workstation", "domain controller", "server")[winVer.product_type - 1]
+	)
 
-winVersion=sys.getwindowsversion()
-winVersionText="{v.major}.{v.minor}.{v.build}".format(v=winVersion)
-if winVersion.service_pack_major!=0:
-	winVersionText+=" service pack %d"%winVersion.service_pack_major
-	if winVersion.service_pack_minor!=0:
-		winVersionText+=".%d"%winVersion.service_pack_minor
-winVersionText+=" %s" % ("workstation","domain controller","server")[winVersion.product_type-1]
 
 def isSupportedOS():
 	# NVDA can only run on Windows 7 Service pack 1 and above
-	return (winVersion.major,winVersion.minor,winVersion.service_pack_major) >= (6,1,1)
+	return getWinVer() >= WIN7_SP1
+
 
 def canRunVc2010Builds():

```

Comment:
I believe this function is not used anywhere and seems to be no longer relevant. Perhaps it can be removed as part of a general clean-up of the `winVersion` module.